eZ/Publish/API/Repository/CompareEngine.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\API\Repository;
+
+use eZ\Publish\API\Repository\Values\Content\VersionDiff\CompareResult;
+use eZ\Publish\SPI\Compare\CompareField;
+
+interface CompareEngine
```

Comment:
naming again, but the other question - isn't this rather an internal thing for implementation details of service used to compare Versions?

Reply:
I would say no - if someone develops it own fieldType with specific data type he must have a way to register engine for comparing them.